nautobot/docs/development/core/kubernetes-job-support.md

```diff
@@ -0,0 +1,412 @@
+# Kubernetes Job Support
+
+Kubernetes Job Support is added in Nautobot v2.4.0. This documentation is a end-to-end development guide on topics from how to set up your local kubernetes cluster with [minikube](https://minikube.sigs.k8s.io/docs/) and how to run a nautobot job in a remote kubernetes job pod.
+
+## Preliminary Setup
+
+First we need to install minikube, go to the offical [get started page](https://minikube.sigs.k8s.io/docs/start/?arch=%2Fmacos%2Farm64%2Fstable%2Fbinary+download) to learn how to download minikube.
+
+Once minikube is downloaded, create and start your minikube cluster with the following command:
+
+```bash
+minikube start
+```
+
+You should see the following output:
+
+```bash
+😄  minikube v1.34.0 on Darwin 14.4 (arm64)
+✨  Using the docker driver based on existing profile
+👍  Starting "minikube" primary control-plane node in "minikube" cluster
+🚜  Pulling base image v0.0.45 ...
+🏃  Updating the running docker "minikube" container ...
+🐳  Preparing Kubernetes v1.31.0 on Docker 27.2.0 ...
+🔎  Verifying Kubernetes components...
+    ▪ Using image gcr.io/k8s-minikube/storage-provisioner:v5
+🌟  Enabled addons: storage-provisioner, default-storageclass
+🏄  Done! kubectl is now configured to use "minikube" cluster and "default" namespace by default
+```
+
+Next we need [`kubectl`](https://kubernetes.io/docs/reference/kubectl/) to interact with the new cluster we just created. See [Kubernetes official documentation](https://kubernetes.io/docs/tasks/tools/#kubectl) on how to download `kubectl` for your respective operating system.
+
+Once you have `kubectl` downloaded and installed, run the following command to ensure the version you installed is up-to-date:
+
+```bash
+kubectl version --client
+```
+
+You should see the following output:
+
+```bash
+Client Version: v1.31.2
+Kustomize Version: v5.4.2
+```
+
+We also need to check if our default service account is enabled to create jobs, we can check the permission by executing the following command:
+
+```bash
+kubectl auth can-i --as=system:serviceaccount:default:default create jobs -n default
+```
+
+If the output from the above command is yes, then you are all good to go. However, if the output is no, then we need to create a rolebinding to grant the default user appropriate permissions. We can achieve this by running the following command:
+
+```bash
+kubectl create rolebinding admin-namespace-default-new --clusterrole=admin --serviceaccount=default:default --namespace=default
+```
+
+This command will assign the admin role to our default serviceaccount in the namespace `default`. Check out kubernetes [RBAC authorization page](https://kubernetes.io/docs/reference/access-authn-authz/rbac/) to learn how to create more granular role and permission assignments.
+
+We can run the `kubectl auth can-i --as=system:serviceaccount:default:default create jobs -n default` again and this time the output should be yes.
```

Comment:
Can we change from mixed "we/you" wording to "you" consistently throughout this file?

```suggestion
You can run the `kubectl auth can-i --as=system:serviceaccount:default:default create jobs -n default` again and this time the output should be `yes`.
```